### PR TITLE
Add 3rd party ssl certificate support

### PR DESCRIPTION
**Pull request by cbosdo**
Part of the logic from `mgr-ssl-cert-setup` has been ported in `shared/ssl/ssl.go` to compute the chain correctly for kubernetes.

### CODE REVIEW

**Comment by cbosdo** [comment]
I just moved back all the SSL logic from `shared` to `uyuniadm/shared`.

**Comment by cbosdo** [comment]
Added unit test for the certificates ordering code

File: uyuniadm/shared/podman/podman.go [hunk 1]
```diff
@@ -126,11 +127,7 @@ func isIpv6Enabled() bool {
 }
 
 func getFileBoolean(file string) bool {
-	out, err := os.ReadFile(file)
-	if err != nil {
-		log.Fatal().Err(err).Msgf("Failed to read file %s", file)
-	}
-	return string(out[:]) != "0"
+	return string(utils.ReadFile(file)) != "0"
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply uyuniadm/shared/podman/podman.go:130]: No, this function helps reading kernel API files that contain either `0` or `1`... so we really want to check the file content.

File: shared/utils/exec.go [hunk 2]
```diff
@@ -52,3 +52,12 @@ func RunCmdOutput(logLevel zerolog.Level, command string, args ...string) ([]byt
 	}
 	return output, err
 }
+
+// Run a command and log its output too.
+func RunCmdLogOutput(logLevel zerolog.Level, command string, args ...string) error {
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply shared/utils/exec.go:57]: I don't understand your question here. I don't think we have a function outputting to both the console and the logger yet. I extract that code from a function that was doing this, but executing it inside the container. This one doesn't know about containers.

**Comment by cbosdo** [comment]
I just pushed a segfault fix and removal of `--ssl-useexisting` flag. Flags grouping in the help would be nice, but for a follow up PR

**Comment by cbosdo** [comment]
Fixes https://github.com/SUSE/spacewalk/issues/22113

**Review by rjmateus — APPROVED** [review]
lgtm

**Comment by cbosdo** [comment]
Added support for 3rd party certificate migration

**Pull request merged**